app/legacy/dao/service_sms_sender_dao.py

```diff
@@ -0,0 +1,113 @@
+            # Exceeded retries or was never retryable. Downstream methods logged this
+            raise NonRetryableError from e
+        return row
+
+    @db_retry
+    @staticmethod
+    async def _get(id: UUID4) -> Row[Any]:
+        legacy_service_sms_senders = metadata_legacy.tables['service_sms_senders']
+        try:
+            stmt = select(legacy_service_sms_senders).where(legacy_service_sms_senders.c.id == id)
+            async with get_read_session_with_context() as session:
+                return (await session.execute(stmt)).one()
+
+        except (IntegrityError, DataError) as e:
+            # These are deterministic and will likely fail again
+            logger.exception('Notification lookup failed: invalid or unexpected data for id: {}', id)
+            raise NonRetryableError('Notification lookup failed: invalid or unexpected data.') from e
+
+        except (OperationalError, InterfaceError, TimeoutError) as e:
+            # Transient DB issues that may succeed on retry
+            logger.warning('Notification lookup failed due to a transient database error for id: {}', id)
+            raise RetryableError('Notification lookup failed due to a transient database error.') from e
+
+        except SQLAlchemyError as e:
+            logger.exception('Unexpected SQLAlchemy error during notification lookup for id: {}', id)
+            raise NonRetryableError('Unexpected SQLAlchemy error during notification lookup.') from e
+
+    @db_retry
+    @staticmethod
+    async def _get_service_default(service_id: UUID4) -> Row[Any]:
+        legacy_service_sms_senders = metadata_legacy.tables['service_sms_senders']
+        try:
+            stmt = (
+                select(legacy_service_sms_senders)
+                .where(legacy_service_sms_senders.c.service_id == service_id)
+                .where(legacy_service_sms_senders.c.is_default)
+            )
+            async with get_read_session_with_context() as session:
+                return (await session.execute(stmt)).one()
+
+        except (IntegrityError, DataError) as e:
+            # These are deterministic and will likely fail again
+            logger.exception('Notification lookup failed: invalid or unexpected data for id: {}', id)
+            raise NonRetryableError('Notification lookup failed: invalid or unexpected data.') from e
+
+        except (OperationalError, InterfaceError, TimeoutError) as e:
+            # Transient DB issues that may succeed on retry
+            logger.warning('Notification lookup failed due to a transient database error for id: {}', id)
+            raise RetryableError('Notification lookup failed due to a transient database error.') from e
+
+        except SQLAlchemyError as e:
+            logger.exception('Unexpected SQLAlchemy error during notification lookup for id: {}', id)
+            raise NonRetryableError('Unexpected SQLAlchemy error during notification lookup.') from e
```

Comment:
copy / paste issue. These are not notification lookups.

Reply:
Fixed.